File: Frontend/react_frontend/src/services/EmployeeService.js
import axios from 'axios';
const URL="http://localhost:9090/api/v1/employees";
class EmployeeService
{
    getEmployees()
    {
        return axios.get(URL)
    }
    createEmployee(employee)
    {
         return axios.post(URL,employee);   
    }
    getEmployeeById(employeeId)
    {
        return axios.get(URL+'/'+employeeId);
    }
    updateEmployee(employeeId,employee)
    {
        return axios.put(URL+'/'+employeeId,employee);
    }
    deleteEmployee(employeeId)
    {
          return axios.delete(URL+'/'+employeeId);  
    }
}
export default new EmployeeService();

File: Backend/Backend/src/main/java/com/aslam/www/service/EmployeeService.java
package com.aslam.www.service;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import com.aslam.www.exception.ResourceNotFoundException;
import com.aslam.www.model.Employee;
import com.aslam.www.repository.EmployeeRepository;
@Service
public class EmployeeService
{
	@Autowired
	EmployeeRepository er;
	public List<Employee> getAllEmployees()
	{
		return er.findAll();
	}
	public Employee createEmployee(Employee employee)
	{
		return er.save(employee);
	}
	public Employee getEmployeeById(Long id)
	{
		return er.findById(id).orElseThrow(()->new ResourceNotFoundException("Id Not Found"));
	}
	public ResponseEntity<Employee> updateEmployee(Long id,Employee employee)
	{
		Employee emp=er.findById(id).orElseThrow(()->new ResourceNotFoundException("Id Not found"));
		emp.setFirstName(employee.getFirstName());
		emp.setLastName(employee.getLastName());
		emp.setEmail(employee.getEmail());
		Employee update=er.save(emp);
		return ResponseEntity.ok(update);
	}
	public ResponseEntity<HttpStatus> deleteEmployee(long id)
	{
		Employee employee=er.findById(id).orElseThrow(()->new ResourceNotFoundException("Id Not Found"));
		er.delete(employee);
		return new ResponseEntity<>(HttpStatus.NO_CONTENT);
	}
}
